core/core.go

```
@@ -38,14 +40,14 @@ func NewCore(pactusBlockStart, polygonOrderStart int, filepaths ...string) *Core
 
 	db, err := database.NewDB(cfg.DBPath)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	pactusC := pactusClient.NewPactusClient()
 	for _, n := range cfg.PacLsn.RPCURLS {
-		err = pactusC.AddClient(n)
+		err = pactusC.AddClient(n) // TODO using example client
 		if err != nil {
-			panic(err) // TODO: must be graceful shutdown
+			log.Print(n) // TODO add logger
```

i think it is better to return the error and handle it in higher level

---

We can't stop our process if only one RPC URL is unhealthy. we need to log it for information and continue adding other RPCs.